solutions/platinum/cses-1742.mdx

```diff
@@ -0,0 +1,246 @@
+---
+id: cses-1742
+source: CSES
+title: Robot Path
+author: Mihnea Brebenel
+---
+
+## Explanation
```

Comment:
This explanation needs to be more clear. Explicitely state that the check function of the binary search is to determine if the number of intersections is equal to `[number of segments] - 1` (or whatever you're doing in your code). 

You need to specify that this is being done with a segment tree. It may be worth linking the editorial to [this problem](https://cses.fi/problemset/task/1740). 

There are also a couple implementation tricks that are worth mentioning. Firstly, any adjacent horizontal-horizontal or vertical-vertical actions can be merged, or is an upper bound on the answer. You should also mention coordinate compression and/or using an ordered set instead. You need to be clear on how your handling a) adjacent segments that intersect at their end points, and b) intersections between segments of the same direction (since `CSES intersection points forbids these types of intersections). Finally, you can directly binary search on the distance, instead of doing two binary searches on the segment, and then the distance.

Reply:
Ok. It's ok now ?